src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+            diff = depth_aligned[i] - depth_aligned[j]
+            cost += (diff**2).mean().sqrt().item()
+
+        if regularizer_strength > 0:
+            prediction, _ = ensemble(depth_aligned, return_uncertainty=False)
+            err_near = (0.0 - prediction.min()).abs().item()
+            err_far = (1.0 - prediction.max()).abs().item()
+            cost += (err_near + err_far) * regularizer_strength
+
+        return cost
+
+    init_min = depth_to_align.reshape(E, -1).min(dim=1).values
+    init_max = depth_to_align.reshape(E, -1).max(dim=1).values
+    init_s = 1.0 / (init_max - init_min).clamp(min=1e-6)
+    init_t = -init_s * init_min
+    init_st = torch.cat((init_s, init_t)).cpu().numpy()
+
+    res = minimize(
+        cost_fn,
+        init_st,
+        method="BFGS",
+        tol=tol,
+        options={"maxiter": max_iter, "disp": False},
+    )
+    s, t = np.split(res.x, 2)
+
+    depth = align_depth(depth, s, t)
+    depth, uncertainty = ensemble(depth, return_uncertainty=output_uncertainty)
+
+    depth_min = depth.min()
+    depth_max = depth.max()
+    depth_range = (depth_max - depth_min).clamp(min=1e-6)
+    depth = (depth - depth_min) / depth_range
+    if output_uncertainty:
+        uncertainty /= depth_range
+
+    return depth, uncertainty  # [1,1,H,W], [1,1,H,W]
+
+
+def colormap(
+    image: Union[np.ndarray, torch.Tensor],
+    cmap: str = "Spectral",
+    bytes: bool = False,
+    force_method: Optional[str] = None,
+) -> Union[np.ndarray, torch.Tensor]:
+    assert force_method in (None, "matplotlib", "custom")
+    if not (torch.is_tensor(image) or isinstance(image, np.ndarray)):
+        raise ValueError("Argument must be a numpy array or torch tensor.")
+
+    def method_matplotlib(image, cmap, bytes=False):
+        try:
+            import matplotlib
+        except ImportError:
+            return None
+
+        arg_is_pt, device = torch.is_tensor(image), None
+        if arg_is_pt:
+            image, device = image.cpu().numpy(), image.device
+
+        if cmap not in matplotlib.colormaps:
+            raise ValueError(
+                f"Unexpected color map {cmap}; available options are: {', '.join(list(matplotlib.colormaps.keys()))}"
+            )
+
+        cmap = matplotlib.colormaps[cmap]
+        out = cmap(image, bytes=bytes)  # [?,4]
+        out = out[..., :3]  # [?,3]
+
+        if arg_is_pt:
+            out = torch.tensor(out, device=device)
+
+        return out
+
+    def method_custom(image, cmap, bytes=False):
+        arg_is_np = isinstance(image, np.ndarray)
+        if arg_is_np:
+            image = torch.tensor(image)
+        if image.dtype == torch.uint8:
+            image = image.float() / 255
+        else:
+            image = image.float()
+
+        if cmap != "Spectral":
+            raise ValueError("Only 'Spectral' color map is available without installing matplotlib.")
+
+        _Spectral_data = (  # Copied from matplotlib/_cm.py
```

Comment:
(nit): Can we write "# Taken from matplotlib/_cm.py" here as "# Copied from ..." has a separate treatment in the library codebase.

Reply:
Fixed